ls1intum/Artemis
# `Programming exercises`: Fix display of submission count for policy and add it to code editor

### Checklist
#### General


#### Server
#### Client

### Motivation and Context


Fix #6520 and also display number of submissions in code editor

### Description

The number of submissions for programming exercises should now be displayed correctly in the client. Additionally it is displayed in the code editor by using a new endpoint to load the number of submissions.
This PR only changes the submission display in the course version of the editor. The exam code editor currently does not show the submission count at all.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise with a submission Policy and with the online editor enabled

1. Log in to Artemis
2. Participate in the programming exercise
3. Check that the submission count increases with each submission
4. Check that the submission policy is correctly applied once the limit is reached

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage



| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| SubmissionPolicyService.java | 91% | ✅                           |
| code-editor-student-container.component.ts | 84% | ✅         |
| header-exercise-page-with-details.component.ts | 98% |  ✅ |

### Screenshots


![Bildschirmfoto 2023-05-25 um 12 24 21](https://github.com/ls1intum/Artemis/assets/38322605/3234d208-0659-48c0-ab88-8cd07138ac8d)
